library/client_events.js

```diff
@@ -114,6 +114,21 @@ module.exports = client => {
             client.createGuild(newMember.guild.id)
             client.log('Created new guild.')
           } else {
+            // if in any practice channel, member has rights to speak in practice room chat (can be muted)
+            let prChan = client.channels.find(chan => chan.name === 'practice-room-chat')
+            if (prChan == null) {
+              client.log('Cannot find #practice-room-chat!')
+            } else if (restwo.permitted_channels.includes(newMember.voiceChannelID)) {
+              prChan.overwritePermissions(newMember.user, { SEND_MESSAGES: true })
+            } else {
+              // remove the permission overwrite when member is no longer in a practice room
+              let existingOverride = prChan.permissionOverwrites.get(newMember.user.id)
```

Comment:
Should we ensure that the permission overwrite doesn't include other overwrites unrelated to `SEND_MESSAGES`?

Reply:
I thought about this previously and couldn't imagine a scenario where we had individual overwrites on #pr-chat not related to this... but I've gone ahead and done this anyway.